src/main/java/io/qdrant/spark/QdrantOptions.java

```diff
@@ -1,38 +1,79 @@
 package io.qdrant.spark;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 
 /** This class represents the options for connecting to a Qdrant instance. */
 public class QdrantOptions implements Serializable {
   public String qdrantUrl;
   public String apiKey;
   public String collectionName;
-  public String embeddingField;
   public String idField;
-  public String vectorName;
-  public int batchSize = 100;
+
+  public int batchSize = 64;
   public int retries = 3;
 
+  // Should've named the option 'vectorField'. But too late now.
+  public String embeddingField;
+  public String vectorName;
+
+  public String[] sparseVectorValueFields;
+  public String[] sparseVectorIndexFields;
+  public String[] sparseVectorNames;
+
+  public String[] vectorFields;
+  public String[] vectorNames;
+
+  public List<String> payloadFieldsToSkip = new ArrayList<String>();
+
   /**
    * Constructor for QdrantOptions.
    *
    * @param options A map of options for connecting to a Qdrant instance.
    */
   public QdrantOptions(Map<String, String> options) {
-    this.qdrantUrl = options.get("qdrant_url");
-    this.collectionName = options.get("collection_name");
-    this.embeddingField = options.get("embedding_field");
-    this.idField = options.get("id_field");
-    this.apiKey = options.get("api_key");
-    this.vectorName = options.get("vector_name");
+    qdrantUrl = options.get("qdrant_url");
+    collectionName = options.get("collection_name");
+    embeddingField = options.getOrDefault("embedding_field", "");
+    idField = options.getOrDefault("id_field", "");
+    apiKey = options.getOrDefault("api_key", "");
+    vectorName = options.getOrDefault("vector_name", "");
+    sparseVectorValueFields = options.getOrDefault("sparse_vector_value_fields", "").split(",");
+    sparseVectorIndexFields = options.getOrDefault("sparse_vector_index_fields", "").split(",");
+    sparseVectorNames = options.getOrDefault("sparse_vector_names", "").split(",");
```

Comment:
probably we can just take the sparse_vector_names and expect the cols to be of type `{indices:[],values:[]}`

since indices and values are created in the same process while generation, it would be easier to form the struct col. 

would reduce configuration complexity during write.

Reply:
[Separate array fields ](https://github.com/qdrant/qdrant-spark/blob/7b228087317963e948d361f3361c927e34e34837/src/test/python/test_qdrant_ingest.py#L52-L63)would be simpler instead of single Struct field.